cloudinit/cmd/main.py

```diff
@@ -348,6 +348,9 @@ def _should_wait_via_user_data(
         )
         return True, "failed to parse user data as yaml"
 
+    if not isinstance(parsed_yaml, dict):
```

Comment:
This works, but I would do it different.

`handlers.type_from_starts_with()` only returned `"text/cloud-config"` because of a bug in the slicing on line 334. We should fix the bug instead. I'm guessing that slicing by the length of the longest value in `INCLUSION_TYPES_MAP` should do it.

As it is, either of the following types will incorrectly return `text/cloud-config`:

`#cloud-config-archive`
`#cloud-config-jsonp`

Reply:
As an alternative, you could also do `.split(maxsplit=1)[0]` to just get the first word

Reply:
@kageurufu , thanks for the suggestion, but the goal of the slice was to avoid parsing processing the entire blob.

Reply:
> the goal of the slice was to avoid parsing processing the entire blob.

@kageurufu's suggestion would accomplish that: split on the first whitespace and return only the first value

Reply:
@holmanb I was more thinking about non-text binary data like some big gzipped blob.